File: ministr-core/tests/fixtures/symbols/sample.c
/**
 * Maximum retry count.
 */
#define MAX_RETRIES 3

/**
 * Application configuration.
 */
struct AppConfig {
    char *name;
    int debug;
};

/**
 * Color options.
 */
enum Color {
    RED,
    GREEN,
    BLUE
};

/**
 * Greet a user by name.
 */
void greet(const char *name) {
    printf("Hello, %s!\n", name);
}

/**
 * Create a new AppConfig.
 */
struct AppConfig new_config(const char *name) {
    struct AppConfig cfg;
    cfg.name = (char *)name;
    cfg.debug = 0;
    return cfg;
}

File: ministr-core/tests/fixtures/symbols/sample.cs
namespace Sample;

/// <summary>Maximum retry count.</summary>
public const int MaxRetries = 3;

/// <summary>A serialization interface.</summary>
public interface ISerializable
{
    string Serialize();
}

/// <summary>Configuration for the application.</summary>
public class AppConfig : ISerializable
{
    public string Name { get; set; }
    public bool Debug { get; set; }

    public AppConfig(string name, bool debug)
    {
        Name = name;
        Debug = debug;
    }

    /// <summary>Check if debug mode is on.</summary>
    public bool IsDebug()
    {
        return Debug;
    }

    public string Serialize()
    {
        return Name;
    }
}

/// <summary>Color options.</summary>
public enum Color
{
    Red,
    Green,
    Blue
}

/// <summary>Greet a user by name.</summary>
public static string Greet(string name)
{
    return $"Hello, {name}!";
}
